library/general/src/modules/Arch.rb

```diff
@@ -446,6 +449,23 @@ def is_kvm
     end
 
     # ************************************************************
+    # zKVM stuff
+
+    # zKVM means KVM on IBM System z
+    # true if zKVM is running
+    #
+    # @return true if we are running on zKVM hypervisor
+    def is_zkvm
+      if @_is_zkvm.nil?
+        # using different check than on x86 as recommended by IBM
+        @_is_zkvm = Yast::WFM.Execute(".local.bash", "egrep 'Control Program: KVM' /proc/sysinfo") == 0
+        @_is_zkvm &&= s390
```

Comment:
maybe it make sense to have different order as you would like do quick check for s390 and then run some script if needed.